lib/web/apiserver.go

```diff
@@ -985,6 +985,7 @@ func (h *Handler) bindDefaultEndpoints() {
 	h.GET("/webapi/scripts/integrations/configure/listdatabases-iam.sh", h.WithLimiter(h.awsOIDCConfigureListDatabasesIAM))
 	h.POST("/webapi/sites/:site/integrations/aws-oidc/:name/deployservice", h.WithClusterAuth(h.awsOIDCDeployService))
 	h.POST("/webapi/sites/:site/integrations/aws-oidc/:name/deploydatabaseservices", h.WithClusterAuth(h.awsOIDCDeployDatabaseServices))
+	h.POST("/webapi/sites/:site/integrations/aws-oidc/:name/listdeployeddatabaseservices", h.WithClusterAuth(h.awsOIDCListDeployedDatabaseService))
```

Comment:
Should this be `GET`?

Reply:
+1

Reply:
We have been using `POST` for most of the integration actions, even when listing things.
https://github.com/gravitational/teleport/blob/04bc1c91e5987e51a8edbfb7a8290239b7207d9e/lib/web/apiserver.go#L984
https://github.com/gravitational/teleport/blob/04bc1c91e5987e51a8edbfb7a8290239b7207d9e/lib/web/apiserver.go#L990
https://github.com/gravitational/teleport/blob/04bc1c91e5987e51a8edbfb7a8290239b7207d9e/lib/web/apiserver.go#L991
https://github.com/gravitational/teleport/blob/04bc1c91e5987e51a8edbfb7a8290239b7207d9e/lib/web/apiserver.go#L995-L997

I see these endpoints as a way to execute some action in the Integration. So, something closer to a RPC over HTTP, in contrast with a more typical REST format.

Reply:
> We have been using POST for most of the integration actions

Hmm, any background on why we decided to do this? Unless there's strong reason, sticking with established conventions is usually the preferred option.

Reply:
I think it started like this because we wanted this RPC-like interface where we could register a single endpoint but then use url path variables to call the AWS OIDC gRPC methods directly.

Things evolved in a different manner, and at this point I would argue that following a more REST-y approach would be better (less surprises).

So, we either start converting them to use GETs where is makes sense or we keep it as is.
No strong opinion on which one to follow.

Reply:
my 2c: keep the existing endpoints as-is for backward compatibility (unless you want to migrate them), but for this PR make the new list deployed db services endpoint GET

Reply:
Switched to `GET`
cc @michellescripts

Reply:
It panics with:
```
panic: 'aws-oidc' in new path '/webapi/sites/:site/integrations/aws-oidc/:name/listdeployeddatabaseservices' conflicts with existing wildcard ':name' in existing prefix '/webapi/sites/:site/integrations/:name'
```

Either we implement a handler that picks the correct flow or we changed it back to POST.
I decided to change it back to a POST, unless someone has a strong opinion on going the other option.

Reply:
This appears to be a limitation of [httprouter](https://github.com/julienschmidt/httprouter), unfortunately 😞 There are workarounds but they aren't great, so differentiating the routes with different methods seems to be the most straightforward solution. It's surprising that `httprouter` doesn't handle this case 🤔